code/modules/pool/pool_structures.dm

```diff
@@ -0,0 +1,151 @@
+				jumper.visible_message("<span class='notice'>[user] climbs up \the [src]!</span>", \
+									 "<span class='notice'>You climb up \the [src] and prepares to jump!</span>")
+				jumper.mobility_flags &= ~MOBILITY_MOVE
+				jumper.Stun(40)
+				jumping = TRUE
+				jumper.layer = RIPPLE_LAYER
+				jumper.pixel_x = 3
+				jumper.pixel_y = 7
+				jumper.dir=8
+				sleep(1)
+				jumper.loc = T
+				addtimer(CALLBACK(src, .proc/dive, jumper), 10)
+
+/obj/structure/pool/Lboard/proc/dive(mob/living/carbon/jumper)
+	switch(rand(1, 100))
+		if(1 to 20)
+			jumper.visible_message("<span class='notice'>[jumper] goes for a small dive!</span>", \
+								 "<span class='notice'>You go for a small dive.</span>")
+			sleep(15)
+			backswim()
+			var/atom/throw_target = get_edge_target_turf(src, dir)
+			jumper.throw_at(throw_target, 1, 1)
+
+		if(21 to 40)
+			jumper.visible_message("<span class='notice'>[jumper] goes for a dive!</span>", \
+								 "<span class='notice'>You're going for a dive!</span>")
+			sleep(20)
+			backswim()
+			var/atom/throw_target = get_edge_target_turf(src, dir)
+			jumper.throw_at(throw_target, 2, 1)
+
+		if(41 to 60)
+			jumper.visible_message("<span class='notice'>[jumper] goes for a long dive! Stay far away!</span>", \
+					"<span class='notice'>You're going for a long dive!!</span>")
+			sleep(25)
+			backswim()
+			var/atom/throw_target = get_edge_target_turf(src, dir)
+			jumper.throw_at(throw_target, 3, 1)
+
+		if(61 to 80)
+			jumper.visible_message("<span class='notice'>[jumper] goes for an awesome dive! Don't stand in \his way!</span>", \
+								 "<span class='notice'>You feel like this dive will be awesome</span>")
+			sleep(30)
+			backswim()
+			var/atom/throw_target = get_edge_target_turf(src, dir)
+			jumper.throw_at(throw_target, 4, 1)
+		if(81 to 91)
+			sleep(20)
+			backswim()
+			jumper.visible_message("<span class='danger'>[jumper] misses \his step!</span>", \
+							 "<span class='userdanger'>You misstep!</span>")
+			var/atom/throw_target = get_edge_target_turf(src, dir)
+			jumper.throw_at(throw_target, 0, 1)
+			jumper.Knockdown(100)
+			jumper.adjustBruteLoss(10)
+
+		if(91 to 100)
+			jumper.visible_message("<span class='notice'>[jumper] is preparing for the legendary dive! Can he make it?</span>", \
+								 "<span class='userdanger'>You start preparing for a legendary dive!</span>")
+			jumper.SpinAnimation(7,1)
+
+			sleep(30)
+			if(prob(75))
+				backswim()
+				jumper.visible_message("<span class='notice'>[jumper] fails!</span>", \
+						 "<span class='userdanger'>You can't quite do it!</span>")
+				var/atom/throw_target = get_edge_target_turf(src, dir)
+				jumper.throw_at(throw_target, 1, 1)
+			else
+				jumper.fire_stacks = min(1,jumper.fire_stacks + 1)
+				jumper.IgniteMob()
+				sleep(5)
+				backswim()
+				jumper.visible_message("<span class='danger'>[jumper] bursts into flames of pure awesomness!</span>", \
+					 "<span class='userdanger'>No one can stop you now!</span>")
```

Comment:
make this give you a good moodlet (and make failing an awesome dive or missing your step give you a negative moodlet?)